README.md

```diff
@@ -54,7 +54,7 @@ Forms are well tested way to reduce the coupling between API consumers and provi
 
 ## Media Type
 
-This extension is to be used as a profile link ([RFC6906](https://tools.ietf.org/html/rfc6906)) in conjunction with a HAL style media type.
+This extension is to be used as a profile link ([RFC6906](https://tools.ietf.org/html/rfc6906)) in conjunction with a HAL style media type. This profile link is used primarily to direct develoers this documentation. API consumers SHOULD NOT rely on the profile being present.
```

Comment:
i'm curious about why this should be called out this way. the `profile` does provide information that can be used programmatically to better understand the expectations of the contents. this can allow a consumer to understand how to parse before looking at the body of the response.

also, `s/develoers/developers`

Reply:
@travi, using the profile link to change processing behavior is more complicated than necessary and leads to surprising failures. The following scenarios are the ones that i am considering.

- a) representation with forms, from response with profile
- b) representation without forms, from response with profile
- c) representation with forms, from response without profile
- d) representation without forms, from response without profile
- e) representation with forms, pulled from an embedded section
- f) representation without forms, pulled from an embedded section
- g) representation with forms, pulled from disk
- h) representation without forms, pulled from disk

A consumer that just looks for the the affordances it needs (ie, the `_forms` property) will accomplish its goal in all scenarios where forms exist. However, a consumer that relies on the profile attribute will **only** accomplish its goal in scenario *a* -- and maybe *e* depending on how it is implemented -- even though it **could** succeed in every scenarios the profile agnostic implementation does.

I am unable to come up with a single scenario in which a profile checking consumer produces a better result than a profile agnostic consumer. Even when they produce the same result, such as scenario *a* the profile checking consumer will be more complicated due to the extra check inherent in that approach.

There is a sense in which this stance is merely a specialization of [Postel's law: Be conservative in what you send, be liberal in what you accept.](https://en.wikipedia.org/wiki/Robustness_principle)

PS: Thanks for the spelling correction.  Will fix that.

Reply:
i understand the limited use case that can be handled by using the `profile` programmatically, but i think the use of `SHOULD NOT` is a bit strong. 

my thought around this is mostly because Ii would normally only expect the producer to respond with the form representation of the resource when the consumer asks for it in the `Accept` header. when this representation is requested, i would expect the producer to be strict about setting the `profile` correctly in the `Content-Type`

this may be an implementation detail that is not shared by all providers that use this spec, but for those that do, i believe that consumers `SHOULD` inspect the media-type in a strict manner.

keep in mind that the rfc says
> to allow clients to learn about additional semantics (constraints, conventions, extensions) that are associated with the resource representation, in addition to those defined by the media type and possibly other mechanisms

i think suggesting that the `profile` is only useful to human developers is misleading related to the intended use of a `profile`

does this help clarify my perspective?

Reply:
@pezra the purpose of profile linking the extension specification is so that a client can _know_ that `_forms` will (or at least is intended to) conform to that specification, as opposed to possibly being an `_forms` node conforming to another, unknown (possibly undocumented) spec. 

This means the client can be confident that the interactions will behave as expected, rather than risking unknown side-effects.

Reply:
I apologize, this got merged on accident. I agree with @travi and @mcintyre321 here. 

I see the inclusion of a `profile` essentially acting as a different `Content-Type`, while acknowledging that it's an "extension".

Inspection of the body in order to determine if `_forms` are present seems like something that would be easier accomplished with dynamic languages, but not a realistic solution for anything that is unmarshalling to a specific type/format based on what the client is expecting back (told via the `Accept` header).

Reply:
i'm ok with leaving this unchanged, though i do think it reduces the interoperability and reliability a little. 

@travi, given a HAL representation with a valid `_forms` element but whose content type omits the profile attribute, should conforming clients ignore the forms entirely?

@mcintyre321, i follow that logic but i prefer the more fault tolerant end of the spectrum. if a HAL representation has a`_forms` property and it contains valid form objects it is almost certainly close enough to this spec that clients could safely use it. 

@jareddellitt, i read the spec as **requiring** consumers to check for the existence of forms. the `_forms` property is optional so any conforming  consumer should accept HAL documents with this profile but without a `_forms` property. the spec also any number of forms on any HAL representation, so conforming consumers should accept that too. if those things are true then the consumer could unmarshal **any** HAL document into that structure, regardless of its profile.

i prefer systems that strive for high availability and fault tolerance. getting the content type wrong is easy, and quite common. the spec, as written, misses an opportunity to improve the ease of use and reliability of HAL forms based APIs.

Reply:
i'm going to lift that last paragraph into an issue of it's very own.